llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -968,7 +968,8 @@ NVPTXTargetLowering::NVPTXTargetLowering(const NVPTXTargetMachine &TM,
   setOperationAction(ISD::CopyToReg, MVT::i128, Custom);
   setOperationAction(ISD::CopyFromReg, MVT::i128, Custom);
 
-  // No FEXP2, FLOG2.  The PTX ex2 and log2 functions are always approximate.
+  setOperationAction(ISD::FEXP2, MVT::f32, Legal);
+  // No FLOG2. The PTX log2 function is always approximate.
```

Comment:
`ex2.approx` is also approximate. If not having precise op is a good enough reason not to make log2 legal, why does the same reasoning not apply to exp2?

Reply:
Perhaps we can add `log2` as well, but that would be for a different commit, no?

Reply:
Added `log2`.